### PR TITLE
SLIM-889, fixed Helpers and added new keys

### PR DESCRIPTION
**Pull request by robinbakkeruscgi**

### CODE REVIEW

File: cucumber-tests-platform-dlms/src/test/java/com/alliander/osgp/cucumber/platform/dlms/Keys.java [hunk 1]
```diff
@@ -78,6 +78,10 @@
     public static final String WITH_LIST_SUPPORTED = "WithListSupported";
     public static final String MODULE_ACTIVE_FIRMWARE_VERSION = "ModuleActiveFirmwareVersion";
 
+    // Keys for SchedulerSteps
```
Comment (rudidevries) [inline cucumber-tests-platform-dlms/src/test/java/com/alliander/osgp/cucumber/platform/dlms/Keys.java]: It is not clear for what section of constants this comment is meant. It looks like it also applies to the daylight savings key. I think there is really no need for such a comment. The purpose of keys is to reuse them, so a comment about where it should be used is counter productive.

File: cucumber-tests-platform/src/test/java/com/alliander/osgp/cucumber/platform/core/Helpers.java [hunk 2]
```diff
@@ -146,16 +146,10 @@ public static DateTime getDate(final Map<String, String> settings, final String
...
-     * <li>now + 3 months
-     * <li>tomorrow - 1 year
-     * <li>yesterday + 2 weeks
-     * <li>today at midday
-     * <li>yesterday at midnight
-     * <li>now at midday + 1 week
-     * </ul>
-     * 
+     *
+     * now + 3 months tomorrow - 1 year yesterday + 2 weeks today at midday
```
Comment (rudidevries) [inline cucumber-tests-platform/src/test/java/com/alliander/osgp/cucumber/platform/core/Helpers.java]: Undo this change. A list should be formatted neatly in javadoc.

File: cucumber-tests-platform/src/test/java/com/alliander/osgp/cucumber/platform/core/TestHelper.java [hunk 3]
```diff
@@ -0,0 +1,21 @@
+package com.alliander.osgp.cucumber.platform.core;
```
Comment (rudidevries) [inline cucumber-tests-platform/src/test/java/com/alliander/osgp/cucumber/platform/core/TestHelper.java]: Missing license header.

File: cucumber-tests-platform/src/test/java/com/alliander/osgp/cucumber/platform/core/TestHelper.java [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+import org.joda.time.DateTime;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class TestHelper {
```
Comment (rudidevries) [inline cucumber-tests-platform/src/test/java/com/alliander/osgp/cucumber/platform/core/TestHelper.java]: TestHelper is about the most awful name you could think of for a class :-P

Also, it seems to test JodaTime functionality. I would assume the JodaTime team is very capable of testing their own code, so we don't need to do that. I think this class can be deleted.

File: cucumber-tests-platform-dlms/src/test/java/com/alliander/osgp/cucumber/platform/dlms/Keys.java [hunk 5]
```diff
@@ -78,6 +78,10 @@
...
+    public static final String SCHEDULED_FOR = "scheduledfor";
```
Comment (rudidevries) [inline cucumber-tests-platform-dlms/src/test/java/com/alliander/osgp/cucumber/platform/dlms/Keys.java]: Don't break the convention of keys starting with an uppercase character.

File: cucumber-tests-platform-dlms/src/test/java/com/alliander/osgp/cucumber/platform/dlms/Keys.java [hunk 6]
```diff
@@ -78,6 +78,10 @@
...
+    public static final String DEVICEOPERATION_TAG = "deviceoperation";
```
Comment (rudidevries) [inline cucumber-tests-platform-dlms/src/test/java/com/alliander/osgp/cucumber/platform/dlms/Keys.java]: Don't break the convention of keys starting with an uppercase character.

File: cucumber-tests-platform/src/test/java/com/alliander/osgp/cucumber/platform/core/Helpers.java [hunk 7]
```diff
@@ -227,25 +221,39 @@ public static DateTime getDateTime(final String dateString) throws Exception {
             switch (what) {
             case "days":
                 retval = retval.plusDays(numberToAddOrSubstract);
+                break;
+            case "minutes":
+                retval = retval.plusMinutes(numberToAddOrSubstract);
```
Comment (rudidevries) [inline cucumber-tests-platform/src/test/java/com/alliander/osgp/cucumber/platform/core/Helpers.java]: Not really something you caused, but I don't like the name 'numberToAddOrSubtract', because it is really only added. Adding a negative number is equal to subtracting a positive number, but the operation is still an addition... 

I you don't want to change it, I will accept that...

File: cucumber-tests-platform-dlms/src/test/java/com/alliander/osgp/cucumber/platform/dlms/Keys.java (hunk 1 above)
Comment (robinbakkeruscgi) [inline cucumber-tests-platform-dlms/src/test/java/com/alliander/osgp/cucumber/platform/dlms/Keys.java]: done

File: cucumber-tests-platform-dlms/src/test/java/com/alliander/osgp/cucumber/platform/dlms/Keys.java (hunk 5 above)
Comment (robinbakkeruscgi) [inline cucumber-tests-platform-dlms/src/test/java/com/alliander/osgp/cucumber/platform/dlms/Keys.java]: done

File: cucumber-tests-platform-dlms/src/test/java/com/alliander/osgp/cucumber/platform/dlms/Keys.java (hunk 6 above)
Comment (robinbakkeruscgi) [inline cucumber-tests-platform-dlms/src/test/java/com/alliander/osgp/cucumber/platform/dlms/Keys.java]: done

File: cucumber-tests-platform/src/test/java/com/alliander/osgp/cucumber/platform/core/Helpers.java (hunk 2 above)
Comment (robinbakkeruscgi) [inline cucumber-tests-platform/src/test/java/com/alliander/osgp/cucumber/platform/core/Helpers.java]: done

File: cucumber-tests-platform/src/test/java/com/alliander/osgp/cucumber/platform/core/Helpers.java (hunk 7 above)
Comment (robinbakkeruscgi) [inline cucumber-tests-platform/src/test/java/com/alliander/osgp/cucumber/platform/core/Helpers.java]: done (renamed to numberToAdd)

File: cucumber-tests-platform/src/test/java/com/alliander/osgp/cucumber/platform/core/TestHelper.java (hunk 3 above)
Comment (robinbakkeruscgi) [inline cucumber-tests-platform/src/test/java/com/alliander/osgp/cucumber/platform/core/TestHelper.java]: done

File: cucumber-tests-platform/src/test/java/com/alliander/osgp/cucumber/platform/core/TestHelper.java (hunk 4 above)
Comment (robinbakkeruscgi) [inline cucumber-tests-platform/src/test/java/com/alliander/osgp/cucumber/platform/core/TestHelper.java]: I renamed this testclass. 
IMO this test is needed because the original Helper.testGetDateTime method failed!

File: cucumber-tests-platform-dlms/src/test/java/com/alliander/osgp/cucumber/platform/dlms/Keys.java [hunk 8]
```diff
@@ -87,4 +87,8 @@
     public static final String RESULT = "Result";
     public static final String DAYLIGHT_SAVINGS_ACTIVE = "DST";
     public static final String DEVIATION = "Deviation";
+
+    public static final String SCHEDULED_FOR = "Scheduledfor";
+    public static final String DEVICEOPERATION_TAG = "Deviceoperation";
+    public static final String KEY_SCHEDULED_DATE = "ScheduledDateKey";
```
Comment (rudidevries) [inline cucumber-tests-platform-dlms/src/test/java/com/alliander/osgp/cucumber/platform/dlms/Keys.java]: Do not use the KEY prefix. The classname is enough of a prefix.

File: cucumber-tests-platform-dlms/src/test/java/com/alliander/osgp/cucumber/platform/dlms/Keys.java [hunk 9]
```diff
@@ -87,4 +87,8 @@
     public static final String RESULT = "Result";
     public static final String DAYLIGHT_SAVINGS_ACTIVE = "DST";
     public static final String DEVIATION = "Deviation";
+
+    public static final String SCHEDULED_FOR = "Scheduledfor";
+    public static final String DEVICEOPERATION_TAG = "Deviceoperation";
```
Comment (rudidevries) [inline cucumber-tests-platform-dlms/src/test/java/com/alliander/osgp/cucumber/platform/dlms/Keys.java]: DeviceOperation is a Integration Layer thing, it don't think it has any meaning within OSGP. This should be moved to the IntegrationKeys of the Integration Layer I think.

Comment (robinbakkeruscgi) [inline cucumber-tests-platform-dlms/src/test/java/com/alliander/osgp/cucumber/platform/dlms/Keys.java]: done

File: cucumber-tests-platform-dlms/src/test/java/com/alliander/osgp/cucumber/platform/dlms/Keys.java (hunk 8 above)
Comment (robinbakkeruscgi) [inline cucumber-tests-platform-dlms/src/test/java/com/alliander/osgp/cucumber/platform/dlms/Keys.java]: done

**Comment by bvdzwet** [comment]
retest this please

**Comment by hannek24** [comment]
retest this please

**Comment by bvdzwet** [comment]
retest this please
